ibis/backends/dask/tests/test_udf.py

```diff
@@ -0,0 +1,483 @@
+import collections
+
+import dask.dataframe as dd
+import numpy as np
+import pandas as pd
+import pandas.testing as tm
+import pytest
+
+import ibis
+import ibis.expr.datatypes as dt
+import ibis.expr.types as ir
+from ibis.udf.vectorized import analytic, elementwise, reduction
+
+from .. import connect
+from ..udf import nullable
+
+
+@pytest.fixture
+def df(npartitions):
+    return dd.from_pandas(
+        pd.DataFrame(
+            {
+                'a': list('abc'),
+                'b': [1, 2, 3],
+                'c': [4.0, 5.0, 6.0],
+                'key': list('aab'),
+            }
+        ),
+        npartitions=npartitions,
+    )
+
+
+@pytest.fixture
+def df2(npartitions):
+    return dd.from_pandas(
+        pd.DataFrame(
+            {
+                'a': np.arange(4, dtype=float).tolist()
+                + np.random.rand(3).tolist(),
+                'b': np.arange(4, dtype=float).tolist()
+                + np.random.rand(3).tolist(),
+                'c': np.arange(7, dtype=int).tolist(),
+                'key': list('ddeefff'),
+            }
+        ),
+        npartitions=npartitions,
+    )
+
+
+@pytest.fixture
+def df3(npartitions):
+    return dd.from_pandas(
+        pd.DataFrame(
+            {
+                'a': np.arange(4, dtype=float).tolist()
+                + np.random.rand(3).tolist(),
+                'b': np.arange(4, dtype=float).tolist()
+                + np.random.rand(3).tolist(),
+                'key': list('ddeefff'),
+            }
+        ),
+        npartitions=npartitions,
+    )
+
+
+@pytest.fixture
+def con(df, df2):
+    return connect({'df': df, 'df2': df2})
+
+
+@pytest.fixture
+def t(con):
+    return con.table('df')
+
+
+@pytest.fixture
+def t2(con):
+    return con.table('df2')
+
+
+@elementwise(input_type=['string'], output_type='int64')
+def my_string_length(series, **kwargs):
+    return series.str.len() * 2
+
+
+@elementwise(input_type=[dt.double, dt.double], output_type=dt.double)
+def my_add(series1, series2, **kwargs):
+    return series1 + series2
+
+
+@reduction(['double'], 'double')
+def my_mean(series):
+    return series.mean()
+
+
+@reduction(input_type=[dt.string], output_type=dt.int64)
+def my_string_length_sum(series, **kwargs):
+    return (series.str.len() * 2).sum()
+
+
+@reduction(input_type=[dt.double, dt.double], output_type=dt.double)
+def my_corr(lhs, rhs, **kwargs):
+    return lhs.corr(rhs)
+
+
+@elementwise([dt.double], dt.double)
+def add_one(x):
+    return x + 1.0
+
+
+@elementwise([dt.double], dt.double)
+def times_two(x):
+    return x * 2.0
+
+
+@analytic(input_type=['double'], output_type='double')
+def zscore(series):
+    return (series - series.mean()) / series.std()
+
+
+@reduction(
+    input_type=[dt.double], output_type=dt.Array(dt.double),
+)
+def quantiles(series, *, quantiles):
+    return list(series.quantile(quantiles))
+
```

Comment:
wouldn't be averse to a line break here

```
-----------
Testing
------------
```

or similar (and one like this for Fixtures), nbd, but improves readability at a glance